teachertool/src/components/HeaderBar.tsx

```diff
@@ -73,20 +77,66 @@ export const HeaderBar: React.FC<HeaderBarProps> = () => {
         );
     };
 
-    const onHomeClicked = () => {
-        pxt.tickEvent(Ticks.HomeLink);
+    function avatarPicUrl(): string | undefined {
+        return teacherTool.userProfile?.idp?.pictureUrl ?? teacherTool.userProfile?.idp?.picture?.dataUrl;
```

Comment:
There was a bug fix that I did for the webapp a while back where I found that dataUrl was no longer giving a photo for Mcrosoft auth. I'm not sure if this still applies, though, because on the upload target I logged in with my Microsoft account and it populated fine.. Not sure what's up with all that, but I wanted to drop this here just in case. https://github.com/microsoft/pxt/pull/9956